src/spikeinterface/core/base.py

```diff
@@ -394,9 +393,12 @@ def to_dict(
             dump_dict["properties"] = {k: self._properties.get(k, None) for k in self._main_properties}
 
         if relative_to is not None:
-            relative_to = Path(relative_to).absolute()
+            relative_to = Path(relative_to).resolve().absolute()
             assert relative_to.is_dir(), "'relative_to' must be an existing directory"
-            dump_dict = _make_paths_relative(dump_dict, relative_to)
+            copy = False if dict_contains_extractors(dump_dict) else True
```

Comment:
I am not sure to follow the logic here. Are we making an inplace modification of extarctors._kwargs ?
If this is the case I think this is not a good idea.
rec.to_dict() should not modify rec itself. Maybe I am missing something.

Reply:
we are. that is the only way `relative_to` can work without `recursive=True`